interfaces/builtin/common_files.go

```diff
@@ -104,6 +104,13 @@ func (iface *commonFilesInterface) validateSinglePath(np string) error {
 	if strings.HasSuffix(np, "/") {
 		return fmt.Errorf(`%q cannot end with "/"`, np)
 	}
+	if strings.HasSuffix(np, "@") {
```

Comment:
This is good but I also think we need to disallow users trying to specify AppArmor variables inside the paths as well - I notice from a historical comment of @jdstrand in https://github.com/canonical/snapd/pull/5845#pullrequestreview-165140654  that he suggested using  something like the following but I don't think it was ever added so perhaps now is a good time to add this?

```
if strings.Contains(np, "@{") {
	return fmt.Errorf(`%q should not use '@{'`, np)
}
```

Reply:
This is already disallowed by `apparmor_sandbox.ValidateNoAppArmorRegexp` (called below after `filepath.Clean`), which prohibits any of `?*[]{}^"`.

https://github.com/olivercalder/snapd/blob/96fc2c65715a91853c899c41c29c53ad7aad5cf8/sandbox/apparmor/apparmor.go#L53

So we can't currently have `@{` because we can't have `{` in the path pattern itself. The only exception is where we append the trailing `{,/,/**}` to the end of the path when we generate the AppArmor snippet.

Reply:
Ah excellent, thanks for letting me know.